WalletWasabi.Tests/RegressionTests/WabiSabi/IntegrationTest.cs

```diff
@@ -0,0 +1,17 @@
+using System.Net.Http;
+using Xunit;
+
+namespace WalletWasabi.Tests.RegressionTests.WabiSabi
+{
+	public abstract class IntegrationTest : IClassFixture<ArenaApiApplicationFactory>
```

Comment:
What is this `IClassFixture<ArenaApiApplicationFactory>`?

Reply:
`IClassFixture<>` is an xunit type used to share expensive-to-create objects between tests. In this case that object is the HTTP server that hosts the API. In other words, we don't want to create and stop a new HTTP server for every single test, instead we create only one HTTP server and make it accessible from all test.

The question is "why would someone want to share such a context with the tests?" after all it is a Web API and it should be reached using an HttpClient, no? This is where Microsoft made it really well because the `Microsoft.AspNetCore.Mvc.Testing` package allows us to configure the server in a different way at a request basis. For example, you can  provide a new `Prison` instance that contains banned coins and test how the API behaves in that case without needing to register a coin, fail to sync (the coin is banned after this point) and then try to use it again.  

Take a look at this article where things are better explained:
https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-5.0

Reply:
that's great, thx.